test_runner/regress/test_pageserver_secondary.py

```diff
@@ -596,43 +596,84 @@ def test_secondary_background_downloads(neon_env_builder: NeonEnvBuilder):
 
         tenant_timelines[tenant_id] = [timeline_a, timeline_b]
 
+    def await_log(pageserver, deadline, expression):
+        """
+        Wrapper around assert_log_contains that waits with a deadline rather than timeout
+        """
+        now = time.time()
+        if now > deadline:
+            raise RuntimeError(f"Timed out waiting for {expression}")
+        else:
+            timeout = int(deadline - now) + 1
+            try:
+                wait_until(timeout, 1, lambda: pageserver.assert_log_contains(expression))  # type: ignore
+            except:
+                log.error(f"Timed out waiting for '{expression}'")
+                raise
+
```

Comment:
Optional: I think it would be better to move this method to `LogUtils` where `assert_log_contains` is

Reply:
Hmm, we can move later if it ends up being useful elsewhere: I think most places we wait for a log just want to use a timeout rather than a deadline